### PR TITLE
Redirected the anonymous methods article from the programming guide

### PR DESCRIPTION
**Pull request by pkulikov**
As agreed in https://github.com/dotnet/docs/pull/13432#discussion_r304244192

### CODE REVIEW

File: docs/csharp/programming-guide/statements-expressions-operators/lambda-expressions.md [hunk 1]
```diff
@@ -35,8 +35,6 @@ Or you can pass it directly as a method argument:
 [!code-csharp-interactive[lambda is argument](~/samples/snippets/csharp/programming-guide/lambda-expressions/Introduction.cs#Argument)]
 
 When you use method-based syntax to call the <xref:System.Linq.Enumerable.Select%2A?displayProperty=nameWithType> method in the <xref:System.Linq.Enumerable?displayProperty=nameWithType> class (as you do in LINQ to Objects and LINQ to XML) the parameter is a delegate type <xref:System.Func%602?displayProperty=nameWithType>. A lambda expression is the most convenient way to create that delegate. When you call the <xref:System.Linq.Queryable.Select%2A?displayProperty=nameWithType> method in the <xref:System.Linq.Queryable?displayProperty=nameWithType> class (as you do in LINQ to SQL) the parameter type is an expression tree type [`Expression<Func<TSource,TResult>>`](<xref:System.Linq.Expressions.Expression%601>). Again, a lambda expression is just a very concise way to construct that expression tree. The lambdas allow the `Select` calls to look similar although in fact the type of object created from the lambda is different.
-
-All restrictions that apply to [anonymous methods](anonymous-methods.md) also apply to lambda expressions.
```
Comment (pkulikov) [inline docs/csharp/programming-guide/statements-expressions-operators/lambda-expressions.md]: A lot of those restrictions are described in the last section of this article about lambda expressions.

File: docs/csharp/language-reference/operators/operator-overloading.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ The following table provides information about overloadability of C# operators:
...
+|[=](assignment-operator.md), [.](member-access-operators.md#member-access-operator-), [?:](conditional-operator.md), [??](null-coalescing-operator.md), [->](pointer-related-operators.md#pointer-member-access-operator--), [=>](lambda-operator.md), [f(x)](member-access-operators.md#invocation-operator-), [as](type-testing-and-conversion-operators.md#as-operator), [checked](../keywords/checked.md), [unchecked](../keywords/unchecked.md), [default](../../programming-guide/statements-expressions-operators/default-value-expressions.md), [delegate](delegate-operator.md), [is](type-testing-and-conversion-operators.md#is-operator), [nameof](nameof.md), [new](new-operator.md), [sizeof](../keywords/sizeof.md), [typeof](type-testing-and-conversion-operators.md#typeof-operator)|These operators cannot be overloaded.|
```
Comment (BillWagner) [inline docs/csharp/language-reference/operators/operator-overloading.md]: This would read better if the operators were code fenced. (Especially with all the commas.)

Yes, that affects all the other rows in this table.

We can address this in this PR, or a later PR, if you prefer.

File: docs/csharp/programming-guide/statements-expressions-operators/lambda-expressions.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/programming-guide/statements-expressions-operators/lambda-expressions.md]: Good analysis. I like this change.

File: docs/csharp/programming-guide/statements-expressions-operators/anonymous-methods.md [hunk 3]
```diff
@@ -1,62 +0,0 @@
...
-  
- Creating anonymous methods is essentially a way to pass a code block as a delegate parameter. Here are two examples:  
-  
- [!code-csharp[csProgGuideDelegates#6](~/samples/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideDelegates/CS/Delegates.cs#6)]
```
Comment (BillWagner) [inline docs/csharp/programming-guide/statements-expressions-operators/anonymous-methods.md]: This should be a separate PR, but it would be good to remove the five samples referenced in this file from the samples repo.

**Review by BillWagner — CHANGES_REQUESTED** [review]
This LGTM @pkulikov 

I chose "request changes" so we resolve the two questions I had before I merge it.

Once we've finished that discussion, this is ready to :shipit:

File: docs/csharp/language-reference/operators/operator-overloading.md (hunk 2 above)
Comment (pkulikov) [reply docs/csharp/language-reference/operators/operator-overloading.md]: > This would read better if the operators were code fenced.

@BillWagner I agree. However if you mean the published version, code-fencing appears to be discarded. For example, take a look at the published version of this same table:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/operator-overloading#overloadable-operators

The row about `&&` and `||` operators contains the sentence:
> However, if a type with the overloaded true and false operators also overloads...

In the source file, the words "true" and "false" are code fenced, but it's not reflected in the published version.

Is it by design?

File: docs/csharp/programming-guide/statements-expressions-operators/anonymous-methods.md (hunk 3 above)
Comment (pkulikov) [reply docs/csharp/programming-guide/statements-expressions-operators/anonymous-methods.md]: @BillWagner done: dotnet/samples#1097

File: docs/csharp/language-reference/operators/operator-overloading.md (hunk 2 above)
Comment (BillWagner) [reply docs/csharp/language-reference/operators/operator-overloading.md]: Hmm. I'll take it up with the engineering team.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
I'll ask about the formatting of operators in links @pkulikov 

In the meantime, I'll :shipit:
